mono/metadata/class.c

```diff
@@ -3381,6 +3381,20 @@ mono_class_is_subclass_of (MonoClass *klass, MonoClass *klassc,
 	return result;
 }
 
+static gboolean 
+mono_interface_implements_interface (MonoClass *interface_implementer, MonoClass *interface_implemented)
+{
+	int i;
+	for (i = 0; i < interface_implementer->interface_count; i++) {
+		MonoClass *ic = interface_implementer->interfaces[i];
```

Comment:
you have to use the getter functions.  `m_class_get_interfaces (interface_implementer) [i]`